lib/poseidon/connection.rb

```diff
@@ -162,8 +162,14 @@ def next_correlation_id
       @correlation_id  += 1
     end
 
-    def raise_connection_failed_error
-      raise ConnectionFailedError, "Failed to connect to #{@host}:#{@port}"
+    def raise_connection_failed_error(ex = nil)
```

Comment:
WDYT about accepting `message` (and not making it optional) so that in the cases above where you don't have an `ex` to give, you can still give something meaningful -- and you can drop the conditional in here.